src/main/java/com/minecolonies/coremod/colony/requestable/SmeltableOre.java

```diff
@@ -0,0 +1,87 @@
+package com.minecolonies.coremod.colony.requestable;
+
+import com.minecolonies.api.colony.requestsystem.factory.IFactoryController;
+import com.minecolonies.api.colony.requestsystem.requestable.IDeliverable;
+import com.minecolonies.api.util.ItemStackUtils;
+import com.minecolonies.coremod.colony.ColonyManager;
+import net.minecraft.item.ItemBlock;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Smeltable requestable.
+ * Delivers a stack of a smeltable ore.
+ */
+public class SmeltableOre implements IDeliverable
+{
+
+    ////// --------------------------- NBTConstants --------------------------- \\\\\\
+    private static final String NBT_COUNT  = "Count";
+    private static final String NBT_RESULT = "Result";
+    ////// --------------------------- NBTConstants --------------------------- \\\\\\
+
+    private final int count;
+
+    private ItemStack result;
+
+    public SmeltableOre(final int count)
+    {
+        this.count = count;
+    }
+
+    public SmeltableOre(final int count, final ItemStack result)
+    {
+        this.count = count;
+        this.result = result;
+    }
+
+    public static NBTTagCompound serialize(final IFactoryController controller, final SmeltableOre food)
+    {
+        final NBTTagCompound compound = new NBTTagCompound();
+        compound.setInteger(NBT_COUNT, food.count);
+
+        if (!ItemStackUtils.isEmpty(food.result))
+        {
+            compound.setTag(NBT_RESULT, food.result.serializeNBT());
+        }
+
+        return compound;
+    }
+
+    public static SmeltableOre deserialize(final IFactoryController controller, final NBTTagCompound compound)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "controller". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1172)